delivery/repositories/project_repository.py

```diff
@@ -49,3 +53,158 @@ def get_project(self, project_name):
+            checksum_path,
+            {path: checksum for sample in project.samples for checksum, path in _sample_checksums(sample) if checksum})
+
+        return checksum_path
+
+    def get_projects(self, runfolder):
+        """
+        Returns a list of RunfolderProject instances, representing all projects found in this runfolder.
+
+        :param runfolder: a Runfolder instance
+        :return: a list of RunfolderProject instances or None if no projects were found
+        :raises: ProjectsDirNotfoundException if the Unaligned directory could not be found in the runfolder
+        """
+        def dir_contains_fastq_files(d):
+            return any(
+                map(
+                    lambda f: f.endswith("fastq.gz"),
+                    self.filesystem_service.list_files_recursively(d)))
+
+        def project_from_dir(d):
+            project = RunfolderProject(
+                name=os.path.basename(d),
+                path=os.path.join(projects_base_dir, d),
+                runfolder_path=runfolder.path,
+                runfolder_name=runfolder.name
+            )
+            project.samples = self.sample_repository.get_samples(project, runfolder)
+            return project
+
+        try:
+            projects_base_dir = os.path.join(runfolder.path, self.PROJECTS_DIR)
+
+            # only include directories that have fastq.gz files beneath them
+            project_directories = filter(
+                dir_contains_fastq_files,
+                self.filesystem_service.find_project_directories(projects_base_dir)
+            )
+
+            return list(map(project_from_dir, project_directories)) or None
+
+        except FileNotFoundError:
+            raise ProjectsDirNotfoundException("Did not find Unaligned folder for: {}".format(runfolder.name))
+
+    def get_report_files(self, project):
+        """
+        Gets the paths to files associated with the supplied project's report. This can be either a MultiQC report or,
+        if no such report was found, a Sisyphus report.
+
+        :param project: a RunfolderProject instance
+        :return: a tuple with the path to the directory containing the report files and a list of the paths to the
+        report files
+        :raises ProjectReportNotFoundException: if no MultiQC or Sisyphus report was found for the project
+        """
+        if self.filesystem_service.exists(self.multiqc_report_path(project)):
+            return self.multiqc_report_files(project)
+        for sisyphus_report_path in self.sisyphus_report_path(project):
+            if self.filesystem_service.exists(sisyphus_report_path):
+                return self.sisyphus_report_files(
+                    self.filesystem_service.dirname(sisyphus_report_path))
+        raise ProjectReportNotFoundException("No project report found for {}".format(project.name))
+
+    @staticmethod
+    def sisyphus_report_path(project):
+        return os.path.join(
```

Comment:
Depending on if you feel like it or not I'd recommend looking into `pathlib` to do this type of thing. It makes it possible to write this type of code in a somewhat more straight forward manor. E.g.
`Path(project.runfolder_path) / "Summary" / project.name / "report.html"` is something I find quite a bit more readable.

Reply:
:+1: I'll look it up

Reply:
I agree that `pathlib` looks nice. However, it seems to me that in that case, we should probably use `pathlib` consistently throughout the service and make sure that there are no unforeseen consequences. If it's ok with you, I'd rather leave this as it is since I think that task can swell more than I'd be prepared for at the moment.